ui: Switch bundling from create react app to webpack 5 (#3390)
scality/metalk8s: ui/README.md

# Metalk8s UI

> ## Prerequisite 

Reviewer: ```suggestion
## Prerequisites
```

Author: Done in 13f451b

node v14+

> npm v6+

Reviewer: So, given the issue with v7, maybe:
```suggestion
npm v6
```

Author: Done in 1b556a5

docker (optional, but recommended)

## Install dependencies

```sh
$ npm install
```

## Run locally with a [metalk8s cluster already running](https://metal-k8s.readthedocs.io/en/latest/developer/running/cluster.html#)

> In `webpack.dev.js` edit the value of ´controlPlaneIP´ and provide your cluster bootstrap node ip. You can get it by running

Reviewer: ```suggestion
In `webpack.dev.js` edit the value of `controlPlaneIP` and provide your cluster bootstrap node's control plane IP. You can get it by running:
```

Author: Done in 13f451b

```sh
# salt-call grains.get metalk8s:control_plane_ip
```

Then run the UI with :

```sh
$ npm run start
```

> This will first starts `<root>/shell-ui` in docker container and expose it on the port 8084 and then will start the Metalk8s UI in dev mode.

Reviewer: ```suggestion
This will first start `<root>/shell-ui` in a Docker container and expose it on port 8084 and then will start the Metalk8s UI in dev mode.
```

Author: Done in 13f451b

> If you don't have docker installed on your machine you can alternatively install `<root>/shell-ui` dependencies via running `npm install` in 

Reviewer: ```suggestion
If you don't have Docker installed on your machine you can alternatively install `<root>/shell-ui` dependencies via running `npm install` in 
```

Author: Done in 13f451b
